workers/unity/Assets/Fps/Scripts/UI/ConnectButton/DefaultConnectionAnimController.cs

```diff
@@ -0,0 +1,60 @@
+﻿using System;
+using UnityEngine;
+using UnityEngine.UI;
+
+namespace Fps
+{
+    public class DefaultConnectionAnimController : MonoBehaviour
+    {
+        public Text informationText;
+        private Animator animator;
+
+        public void Awake()
+        {
+            ConnectionStateReporter.OnConnectionStateChange += OnConnectionStateChange;
+            animator = GetComponentInChildren<Animator>();
+        }
+
+        private void OnConnectionStateChange(ConnectionStateReporter.State state, string information)
+        {
+            informationText.text = information;
+
+            switch (state)
+            {
+                case ConnectionStateReporter.State.None:
+                    break;
+                case ConnectionStateReporter.State.GettingDeploymentList:
+                    break;
+                case ConnectionStateReporter.State.DeploymentListAvailable:
+                    break;
+                case ConnectionStateReporter.State.FailedToGetDeploymentList:
+                    break;
+                case ConnectionStateReporter.State.Connecting:
```

Comment:
You can utilize switch-case fall throughs to condense this down a bit.

I.e. - 

```
case ConnectionStateReporter.State.None:
case ConnectionStateReporter.State.GettingDeploymentList:
case ConnectionStateReporter.State.DeploymentListAvailable:
case ConnectionStateReporter.State.FailedToGetDeploymentList:
case ConnectionStateReporter.State.Connecting:
	break;
```

Basically if there is no statement under a case it will fall through to the next one.